Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -639,6 +639,118 @@ public AndConstraint<TAssertions> BeEquivalentTo(IEnumerable expectation, string
+        /// </remarks>
+        /// <param name="because">
+        /// An optional formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the
+        /// assertion is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotContainEquivalentOf<TExpectation>(TExpectation unexpected, string because = "",
+            params object[] becauseArgs)
+        {
+            return NotContainEquivalentOf(unexpected, config => config, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that a collection of objects does not contain any object equivalent to another object.
+        /// </summary>
+        /// <remarks>
+        /// Objects within the collection are equivalent to the expected object when both object graphs have equally named properties with the same
+        /// value, irrespective of the type of those objects. Two properties are also equal if one type can be converted to another
+        /// and the result is equal.
+        /// Notice that actual behavior is determined by the global defaults managed by <see cref="AssertionOptions"/>.
+        /// </remarks>
+        /// <param name="config">
+        /// A reference to the <see cref="EquivalencyAssertionOptions{TSubject}"/> configuration object that can be used
+        /// to influence the way the object graphs are compared. You can also provide an alternative instance of the
+        /// <see cref="EquivalencyAssertionOptions{TSubject}"/> class. The global defaults are determined by the
+        /// <see cref="AssertionOptions"/> class.
+        /// </param>
+        /// <param name="because">
+        /// An optional formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the
+        /// assertion is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotContainEquivalentOf<TExpectation>(TExpectation unexpected, Func<EquivalencyAssertionOptions<TExpectation>,
+            EquivalencyAssertionOptions<TExpectation>> config, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(config, nameof(config));
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:collection} not to contain equivalent of {0}{reason}, ", unexpected)
```

Comment:
When using `WithExpectation` it should be cleared afterwards using 
```c#
.FailWith(...)
.Then
.ClearExpectation();
```

Otherwise if using

```c#
using (new AssertionScope())
{
    collection.Should().NotContainEquivalentOf(another); // assume this fails
    collection.Should().HaveCount(4); // <-- this will contain the WithExpectation from NotContainEquivalentOf
}
```

Reply:
Since I have only one `FailWith` here, I'd better insert it just there directly. Is this also applied to the second place I use `WithExpectation` under `AssertionScope`? Or under assertion scope `ClearExpectation` is not needed?

Reply:
(I assume that under `new AssertionScope()` it is not needed, since it is a brand new scope, but asking to make sure)

Reply:
I haven't tried it out, but I think creating a new `AssertionScope` might not produce the intended failure message when combing multiple failures.
If my example on how to rewrite it works, I would just go with that.